docs/protocol/extension_logging.md

```diff
@@ -0,0 +1,198 @@
+
+A successful log setting request is indicated by a 200 HTTP status
+code. The response object, identified as $log_setting_response, is
+returned in the HTTP body for every successful log setting request.
+
+```
+$log_setting_response =
+{
+  $log_setting, ...
+}
+
+$log_setting = $string : $string | $boolean | $uint32
+```
+
+Each $log_setting JSON describes a “name”/”value” pair, where the “name” is
+the $string representation of the log setting and the “value” is a $string, $bool, or $uint32 representation of the
+setting value. Currently, the following log settings are defined:
+
+- "log_file" : a $string parameter defining the file where the log outputs will be saved. If not specified,
+log outputs will stream to the console.
+
+- "log_info" : a $boolean parameter that controls whether the Triton server logs INFO level messages. 
+
+- "log_warnings" : a $boolean parameter that controls whether the Triton server logs WARNING level messages. 
+
+- "log_errors" : a $boolean parameter that controls whether the Triton server logs ERROR level messages. 
+
+- "log_verbose_level" : a $uint32 parameter that controls whether the Triton server outputs verbose messages
+of varying degrees. This value can be any integer >= 0. If "log_verbose_level" is 0, no verbose messages
+will be output by the Triton server. If "log_verbose_level" is 1, level 1 verbose messages will be output
+by the Triton server. If "log_verbose_level" is 2, the Triton server will output all verbose messages of 
+level <= 2, etc.
+
+- "log_format" : a $string parameter that controls the format of Triton server log messages. There are currently
+2 formats: "default" and "ISO8601".
+
+
+### Log Setting Response JSON Error Object
+
+A failed log setting request will be indicated by an HTTP error status
+(typically 400). The HTTP body will contain a
+$log_setting_error_response object.
+
+```
+$log_setting_error_response =
+{
+  "error": $string
+}
+```
+
+- “error” : The descriptive message for the error.
+
+#### Log Setting Request JSON Object
+
+A log setting request is made with a HTTP POST to
+the logging endpoint. In the corresponding response, the HTTP body contains the
+response JSON. A successful request is indicated by a 200 HTTP status code.
+
+The request object, identified as $log_setting_request must be provided in the HTTP
+body.
+
+```
+$log_setting_request =
+{
+  $log_setting, ...
+}
+```
+
+When a $log_setting JSON is received (defined above), only the specified
+settings will be updated.
+
+### Example Usage
+The logging protocol extension can be invoked using the curl library in the following manner (assuming 
+a Triton server is running at `localhost:8000`):
+```
+curl -s -w '%{http_code}\n' -d'{"log_verbose_level":1}' localhost:8000/v2/logging
+```
+This command should return a $log_setting_response JSON object with the following format:
+```
+{"log_file":"","log_info":true,"log_warnings":true,"log_errors":true,"log_verbose_level":1,"log_format":"default"}200
```

Comment:
Will the status code immediately follow the response body with no newline/formatting like this? `{body}200`

Reply:
Given the command I provided, yes. But I can add an additional '\n' before the http code which results in the output looking like this:
`{"log_file":"","log_info":true,"log_warnings":true,"log_errors":true,"log_verbose_level":1,"log_format":"default"}
200
`
Do you like this output format better?

Reply:
Just checking whether that was a typo or not. I like the newline output, I think it's easier for users to read in doc. 👍